src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/sdpa.cpp

```diff
@@ -22,60 +23,281 @@ namespace attn {
+        LOG_INFO("  Number of heads: " << params.num_heads);
+        LOG_INFO("  Sequence length: " << params.sequence_length);
+        LOG_INFO("  Head dimension: " << params.head_dim);
+        LOG_INFO("  Data type: " << params.data_type);
+        LOG_INFO("  Has bias: " << (params.has_bias ? "Yes" : "No"));
+        LOG_INFO("  MatMul1 transpose_a: " << matmul1_transpose_a << ", transpose_b: " << matmul1_transpose_b);
+        LOG_INFO("  MatMul2 transpose_a: " << matmul2_transpose_a << ", transpose_b: " << matmul2_transpose_b);
+        LOG_INFO("  Total parameters: " << (params.batch_size * params.num_heads * params.sequence_length *
+                                            params.head_dim));
+    } catch (const std::exception& e) {
+        LOG_DEBUG("Error extracting attention parameters: " << e.what());
+        // Return default/empty parameters on error
+        params = AttentionParams{};
+    }
+
+    return params;
+}
 
-    auto sdpa = opp::wrap_type<ov::op::v13::ScaledDotProductAttention>(
-        {opp::any_input(), opt_rshp_k, opt_rshp_v, opp::any_input(), opp::any_input()});
-    auto trans = opp::wrap_type<ov::op::v1::Transpose>({sdpa, opp::any_input()});
-    auto reshape = opp::wrap_type<ov::op::v1::Reshape>({trans, opp::any_input()});
+/*
+    SDPA Pattern:
+            Convert
+                \       /
+                 Concat
+                    |
+                Unsqueeze
+                    |
+                Broadcast   Convert
+                    |       \       /
+                Reshape       Concat
+        \           /           |
+            MatMul           Unsqueeze
+    \       /                   |
+       Add                   Broadcast
+        |                       |
+     Softmax                Reshape
+            \               /
+                  MatMul
+                    |
+                Transpose
+                    |
+                Reshape
+                    |
+*/
+
+SDPA::SDPA(const std::shared_ptr<ov::npuw::online::Snapshot>& snapshot, const std::string& isol_tag) {
+    auto convert1 = opp::wrap_type<ov::op::v0::Convert>({opp::any_input()});
+    auto concat1 = opp::wrap_type<ov::op::v0::Concat>({convert1, opp::any_input()});
+    auto unsqueeze1 = opp::wrap_type<ov::op::v0::Unsqueeze>({concat1, opp::any_input()});
+    auto broadcast1 = opp::wrap_type<ov::op::v3::Broadcast>({unsqueeze1, opp::any_input()});
+    auto reshape1 = opp::wrap_type<ov::op::v1::Reshape>({broadcast1, opp::any_input()});
+
+    auto convert2 = opp::wrap_type<ov::op::v0::Convert>({opp::any_input()});
+    auto concat2 = opp::wrap_type<ov::op::v0::Concat>({convert2, opp::any_input()});
+    auto unsqueeze2 = opp::wrap_type<ov::op::v0::Unsqueeze>({concat2, opp::any_input()});
+    auto broadcast2 = opp::wrap_type<ov::op::v3::Broadcast>({unsqueeze2, opp::any_input()});
+    auto reshape2 = opp::wrap_type<ov::op::v1::Reshape>({broadcast2, opp::any_input()});
+
```

Comment:
these should be optional like in the original pattern, e.g. Phi3.5 doesn't have these in the model I've been looking at.
the old pattern handled both cases